src/engines/parachain_engine.rs

```diff
@@ -80,35 +80,53 @@ pub fn build_parachain(path: &Option<PathBuf>) -> anyhow::Result<()> {
 #[cfg(test)]
 mod tests {
 	use super::*;
+	use anyhow::{Error, Result};
 	use std::{env::current_dir, fs};
 
-	#[test]
-	fn test_instantiate_template_dir_base() -> Result<(), Box<dyn std::error::Error>> {
-		let temp_dir = tempfile::tempdir()?;
+	fn setup_template_and_instantiate() -> Result<tempfile::TempDir, Error> {
+		let temp_dir = tempfile::tempdir().expect("Failed to create temp dir");
 		let config = Config {
 			symbol: "DOT".to_string(),
 			decimals: 18,
-			initial_endowment: "1000000000000000000000000".to_string(),
+			initial_endowment: "1000000".to_string(),
 		};
-		let result: Result<()> = instantiate_base_template(temp_dir.path(), config);
+		let result: anyhow::Result<()> = instantiate_base_template(temp_dir.path(), config);
 		assert!(result.is_ok());
+		Ok(temp_dir)
+	}
+
+	#[test]
+	fn test_parachain_instantiate_base_template() -> Result<(), Error> {
+		let temp_dir =
+			setup_template_and_instantiate().expect("Failed to setup template and instantiate");
 
 		// Verify that the generated chain_spec.rs file contains the expected content
 		let generated_file_content =
-			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))?;
+			fs::read_to_string(temp_dir.path().join("node/src/chain_spec.rs"))
+				.expect("Failed to read file");
 		assert!(generated_file_content.contains("DOT"));
 		assert!(generated_file_content.contains("18"));
```

Comment:
There can be any number of `18`s in the file. A part of a key for instance. Please improve this. 
`tokenDecimals: 18` is a much better check.

Reply:
Done.